jax/_src/state/discharge.py

```diff
@@ -247,10 +247,21 @@ def index_array(x, indexers):
       continue
     if indexer is None:
       continue
+
+    if all(isinstance(i, indexing.Slice) for i in indexer.indices):
+      t = []
+      for i in indexer.indices:
+        if isinstance(i, indexing.Slice):
```

Comment:
Can we convert any scalar indexer `i` to a single-element slice and then do `squeeze`?